File: Server Frontend/core/src/com/csc480/game/Engine/Model/Player.java
package com.csc480.game.Engine.Model;

/**
 * A simple data structure to hold playerData
 */
public class Player {
    public String name;
    //public int position;
    public char[] tiles;
    public boolean turn;
    public int score;
    public String team;
    public boolean isAI;

    public Player(){
        name = "Not Assigned";
        //position = 0;
        tiles = new char[]{0,0,0,0,0,0,0};
        turn = false;
        score = 0;
        team = "Not Assigned";
        isAI = false;
    }
}

File: Server Backend/entities/Player.js
'use strict'

function Player(id, isAI, position) {
  this.id = id
  this.isAI = isAI
  this.position = position
  this.score = 0
  this.isTurn = false
  this.tiles = []
  this.link = null
  this.name = null
  this.team = null
  // this.listenForIncoming()
}

Player.prototype.addInformation = function(name, team, link) {
  this.name = name
  this.team = {
    name: team.name,
    link: team.link
  }
  this.link = link
  return true
}

Player.prototype.resetScore = function() {
  this.score = 0
}

Player.prototype.updateScore = function(score) {
  this.score += score
}

Player.prototype.injectData = function(data) {
  this.tiles = data.tiles
  this.isTurn = data.isTurn
}

Player.prototype.setTiles = function(tiles) {
  this.tiles = tiles
}

Player.prototype.data = function() {
  return {
    name: this.name,
    team: 'Green',
    // team: this.team.name,
    position: this.position,
    isTurn: this.isTurn,
    isAI: this.isAI,
    score: this.score,
    tiles: this.tiles

  }
}

module.exports = function(id, socket, isAI, position) {
  return new Player(id, socket, isAI, position)
}
